File: Solutions/0290- Word Pattern/Solution.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        my_dict = dict()
        s = s.split(" ")

        if len(pattern) != len(s):
            return False

        for i in range(0, len(s)):
            if pattern[i] not in my_dict:
                if s[i] in my_dict.values():
                    return False
                my_dict[pattern[i]] = s[i]
            else:
                if my_dict[pattern[i]] != s[i]:
                    return False

        return True

File: Solutions/0290- Word Pattern/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean wordPattern(String pattern, String s) {
        Map<Character, String> map = new HashMap<>();

        var sArray = s.split(" ");
        
        if (pattern.length() != sArray.length)
            return false;

        for (int i = 0; i < pattern.length(); i++) {
            char ch = pattern.charAt(i);
            if (!map.containsKey(ch)) {
                if (map.containsValue(sArray[i]))
                    return false;
                map.put(ch, sArray[i]);
            }
            else {
                if (!map.get(ch).equals(sArray[i]))
                    return false;
            }
        }

        return true;


    }
}